lib/rspec/matchers/built_in/change.rb

```diff
@@ -387,8 +387,9 @@ def changed?
           #
           # Note that it is not sufficient to only check the hashes; it is
           # possible for two values to be unequal (and of different classes)
-          # but to return the same hash value.
-          @actual_before != @actual_after || @before_hash != @actual_after.hash
+          # but to return the same hash value. Also, some objects may change
+          # their hash after being compared with `==`/`!=`.
+          @before_hash != @actual_after.hash || @actual_before != @actual_after
```

Comment:
Can we restore this line to its original ordering, but cache the hash value just after line 375
e.g.
```
@actual_before != @actual_after || @before_hash != @actual_hash
```

Reply:
That works as well, will update shortly.
Any objections to using a local variable for `actual_hash`?

Reply:
I'd prefer it if we captured the hash just after the variable, (to minimise the chance of it changing) so it needs to be an ivar from that perspective

Reply:
Works either way. But all ivars is more visually appealing. Changed.

Reply:
Sorry to be a pain, but as per the other comment I would like it moved up to the `perform_change` method, as this would be the first chance to grab the hash and guards against mutation happening elsewhere!